src/Storages/ObjectStorage/DataLakes/DeltaLake/TableSnapshot.cpp

```diff
@@ -381,8 +384,23 @@ size_t TableSnapshot::getVersion() const
     return kernel_snapshot_state->snapshot_version;
 }
 
-bool TableSnapshot::update()
+void TableSnapshot::updateSettings(const DB::ContextPtr & context)
 {
+    const auto & settings = context->getSettingsRef();
+    enable_expression_visitor_logging = settings[DB::Setting::delta_lake_enable_expression_visitor_logging];
+
+    if (settings[DB::Setting::delta_lake_snapshot_version].value != LATEST_SNAPSHOT_VERSION)
```

Comment:
Need to handle all negative numbers now :(

Reply:
It is handled right below.